sql/mysql/sqlspec.go

```diff
@@ -27,12 +27,12 @@ type doc struct {
 
 // evalSpec evaluates an Atlas DDL document into v using the input.
 func evalSpec(p *hclparse.Parser, v any, input map[string]cty.Value) error {
-	var d doc
-	if err := hclState.Eval(p, &d, input); err != nil {
-		return err
-	}
 	switch v := v.(type) {
 	case *schema.Realm:
+		var d doc
+		if err := hclState.Eval(p, &d, input); err != nil {
+			return err
+		}
```

Comment:
Move this after the switch to avoid duplicate code (here, in case of schema.Schema and in default case

Reply:
The code is different in the cases.